pkg/virtctl/root.go

```diff
@@ -36,6 +36,29 @@ func NewVirtctlCommand() *cobra.Command {
 			fmt.Fprint(cmd.OutOrStderr(), cmd.UsageString())
 		},
 	}
+	completionCmd := &cobra.Command{
+		Use:   "completion",
+		Short: "Generates completion scripts for the specified shell",
+		Args:  cobra.MaximumNArgs(1),
```

Comment:
You can use `cobra.ExactArgs(1)`, then you don't have tho check the length of the arguments.

Reply:
You can use `		ValidArgs: []string{"bash", "zsh"},`.